frontend/src/components/transactions/ActionsList.tsx

```diff
@@ -1,31 +1,222 @@
-    let actionRows = actions.map((action, actionIndex) => (
-      <ActionRow
-        key={transaction.hash + actionIndex}
-        action={action}
-        transaction={transaction}
-        viewMode={viewMode}
-      />
-    ));
-
-    if (reversed) {
+    const { actions, transaction, viewMode, reversed, TD } = this.props;
+    let actionRows;
+    let batch = false;
+
+    if (actions.length === 1) {
+      if (TD) {
+        actionRows = actions.map((action, actionIndex) => (
+          <ActionRowBatch
+            key={transaction.hash + actionIndex}
+            action={action}
+            transaction={transaction}
+            viewMode={viewMode}
+          />
+        ));
+      } else {
+        actionRows = actions.map((action, actionIndex) => (
+          <ActionRow
+            key={transaction.hash + actionIndex}
+            action={action}
+            transaction={transaction}
+            viewMode={viewMode}
+          />
+        ));
+      }
+    } else {
+      batch = true;
+      actionRows = actions.map((action, actionIndex) => (
+        <ActionRowBatch
+          key={transaction.hash + actionIndex}
+          action={action}
+          transaction={transaction}
+          viewMode={viewMode}
+        />
+      ));
+    }
+
+    if (reversed && actionRows) {
       actionRows.reverse();
     }
 
-    return <>{actionRows}</>;
+    return (
+      <>
+        {batch && !TD ? (
+          <Row noGutters className={`action-${viewMode}-row mx-0`}>
+            <Col xs="auto">
+              <img
+                src="/static/images/icon-m-batch.svg"
+                className="action-row-img"
+              />
+            </Col>
+            <Col className="action-row-details">
+              <Row noGutters>
+                <Col md="8" xs="7">
+                  <Row noGutters>
+                    <Col className="action-row-title">Batch Transaction</Col>
+                  </Row>
+                  <Row noGutters>
+                    <Col className="action-row-text">
+                      by <AccountLink accountId={transaction.signerId} />
+                    </Col>
+                  </Row>
+                </Col>
+                <Col md="4" xs="5" className="ml-auto text-right">
+                  <Row>
+                    <Col className="action-row-txid">
+                      <TransactionLink transactionHash={transaction.hash} />
+                    </Col>
+                  </Row>
+                  <Row>
+                    <Col className="action-row-timer">
+                      <span className="action-row-timer-status">
+                        <ExecutionStatus status={transaction.status} />
+                      </span>{" "}
+                      <Timer time={transaction.blockTimestamp} />
+                    </Col>
+                  </Row>
+                </Col>
+              </Row>
+            </Col>
+          </Row>
+        ) : null}
+        {actionRows}
```

Comment:
I suggest nesting the batched rows in terms of HTML structure. This way you will have better control over indentation (you just set margin for the block and no need to apply `margin-left` to individual lines) and other styling.